core/ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/texteditor/TextEditorPresenter.java

```diff
@@ -266,16 +279,72 @@ public void action() {
     }
 
     private void setupFileContentUpdateHandler() {
+
+        resourceChangeHandler = generalEventBus.addHandler(ResourceChangedEvent.getType(), new ResourceChangedEvent.ResourceChangedHandler() {
+            @Override
+            public void onResourceChanged(ResourceChangedEvent event) {
+                final ResourceDelta delta = event.getDelta();
+                final Resource resource = delta.getResource();
+
+                switch (delta.getKind()) {
+                    case ADDED:
+                        onResourceCreated(delta);
+                        break;
+                    case REMOVED:
+                        onResourceRemoved(resource);
+                        break;
+                    case UPDATED:
+                        onResourceUpdated(delta);
+                }
```

Comment:
As for me the break statement in the last case is redundant. The same for the default case.